src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -154,9 +154,20 @@ public Integer getContingencyCount(List<UUID> ids, UUID networkUuid, String vari
     }
 
     @Transactional(readOnly = true)
-    public List<Contingency> exportContingencyList(UUID id, UUID networkUuid, String variantId) {
+    public ContingencyListExportResult exportContingencyList(List<UUID> ids, UUID networkUuid, String variantId) {
         Network network = getNetworkFromUuid(networkUuid, variantId);
-        return getContingencies(findContingencyList(id, network), network);
+        List<Contingency> contingencies = new ArrayList<>();
+        List<UUID> notFoundIds = new ArrayList<>();
+
+        ids.forEach(id -> {
+            try {
+                PersistentContingencyList contingencyList = findContingencyList(id, network);
+                contingencies.addAll(getContingencies(contingencyList, network));
+            } catch (ResponseStatusException e) {
+                notFoundIds.add(id);
+            }
+        });
```

Comment:
why not use getAnyContingencyList to have an optional instead of an exception ?

Reply:
not sure what you mean , refactor `findContingencyList` so it does not throw an exception, and replace it with an optional ?